src/main/scala/sbtdatabricks/DatabricksPlugin.scala

```diff
@@ -64,29 +64,33 @@ object DatabricksPlugin extends AutoPlugin {
       }
     }.flatMap(c => c)
   }
-  
+
   private lazy val dbcClasspath = Def.task {
     ((Keys.`package` in Compile).value +: (managedClasspath in Runtime).value.files
       ).filterNot(_.getName startsWith "scala-")
   }
 
   private val dbcFetchClusters = taskKey[Seq[Cluster]]("Fetch all available clusters.")
-  
+
   private lazy val uploadImpl: Def.Initialize[Task[Seq[UploadedLibrary]]] = Def.taskDyn {
     val client = dbcApiClient.value
     val folder = dbcLibraryPath.value
+    val existing = existingLibraries.value
+    val existingSnapshots = existing.filter(_.name.contains("-SNAPSHOT"))
     val classpath = dbcClasspath.value
     var deleteMethodFinished = false
-    val deleteMethod = client.deleteLibraries(existingLibraries.value)
+    val deleteMethod = client.deleteLibraries(existingSnapshots)
     deleteMethodFinished ||= deleteMethod
     if (deleteMethodFinished) {
       Def.task {
-        val uploaded = classpath.map { jar =>
+        // Either upload the newer SNAPSHOT versions, or everything, because they don't exist yet.
+        val toUpload = classpath.toSet -- existing.map(_.jar) ++ existingSnapshots.map(_.jar)
```

Comment:
If lib-1.1.1 and lib-1.2.1 are both of on the classpath the cluster is not usable.